src/snowflake/snowpark/_internal/ast_utils.py

```diff
@@ -859,3 +879,102 @@ def build_udf(
         t = ast.kwargs.add()
         t._1 = k
         build_expr_from_python_val(t._2, v)
+
+
+def build_udtf(
+    ast: proto.Udtf,
+    handler: Union[Callable, Tuple[str, str]],
+    output_schema: Union[
+        StructType, Iterable[str], "PandasDataFrameType"  # noqa: F821
+    ],  # noqa: F821
+    input_types: Optional[List[DataType]],
+    name: Optional[str],
+    stage_location: Optional[str] = None,
+    imports: Optional[List[Union[str, Tuple[str, str]]]] = None,
+    packages: Optional[List[Union[str, ModuleType]]] = None,
+    replace: bool = False,
+    if_not_exists: bool = False,
+    parallel: int = 4,
+    max_batch_size: Optional[int] = None,
+    strict: bool = False,
+    secure: bool = False,
+    external_access_integrations: Optional[List[str]] = None,
+    secrets: Optional[Dict[str, str]] = None,
+    immutable: bool = False,
+    comment: Optional[str] = None,
+    *,
+    statement_params: Optional[Dict[str, str]] = None,
+    source_code_display: bool = True,
+    is_permanent: bool = False,
+    session=None,
+    **kwargs,
+):
+    """Helper function to encode UDTF parameters (used in both regular and mock UDFRegistration)."""
+    # This is the name the UDF is registered to. Not the name to display when unaparsing, that name is captured in callable.
+
+    if name is not None:
+        _set_fn_name(name, ast)
+
+    build_proto_from_callable(
+        ast.handler, handler, session._ast_batch if session is not None else None
+    )
+
+    if output_schema is not None:
+        if isinstance(output_schema, DataType):
+            output_schema._fill_ast(ast.output_schema.udtf_schema__type.return_type)
+        elif isinstance(output_schema, Sequence) and all(
+            isinstance(el, str) for el in output_schema
+        ):
+            ast.output_schema.udtf_schema__names.schema.extend(output_schema)
+        else:
+            raise ValueError(f"Can not encode {output_schema} to AST.")
+
+    # if return_type is not None:
```

Comment:
Commented-out code is hard to read. Is this functionality disabled because it's supposed to be reenabled later? Is it pseudo-code for what the code below actually does?

I suggest deleting the code and, if appropriate, leaving behind a comment about what's missing (if anything), what's feasible here (if it provides useful context for later iterations) etc.

Reply:
let me delete this piece here! I think this a lapse on my end, thanks for spotting!